### PR TITLE
Update drag mapping

### PR DESCRIPTION
**Pull request by imurchie**
Some clients send a different request.

See https://github.com/appium/appium/issues/7559

### CODE REVIEW

File: lib/commands/gesture.js [hunk 1]
```diff
@@ -142,15 +148,25 @@ helpers.handleScroll = async function (gestures) {
...
-
-  // get drag data
-  let pressCoordinates = await this.getCoordinates(press);
-  let duration = (parseInt(wait.options.ms, 10) / 1000);
-  let moveToCoordinates = await this.getCoordinates(moveTo);
+  let pressCoordinates;
+  let duration;
+  let moveToCoordinates;
+  if (gestures.length === 4) {
+    let press = gestures[0];
```
Comment (mykola-mokhnach) [inline lib/commands/gesture.js]: can we use [fail-sort destruction](http://es6-features.org/#FailSoftDestructuring) here?

File: lib/commands/gesture.js [hunk 2]
```diff
@@ -59,13 +59,19 @@ commands.nativeClick = async function (el) {
...
-      gestures.length === 4 &&
+  if (gestures.length === 4 &&
       gestures[0].action === 'press' &&
       gestures[1].action === 'wait' &&
       gestures[2].action === 'moveTo' &&
-      gestures[3].action === 'release'
-  );
+      gestures[3].action === 'release') {
+    return true;
+  } else if (gestures.length === 3 &&
```
Comment (jlipps) [inline lib/commands/gesture.js]: still seems wrong to me that something without any moveTo can be a drag. are we sure we don't want to just fix the clients?

File: lib/commands/gesture.js [hunk 3]
```diff
@@ -142,15 +148,25 @@ helpers.handleScroll = async function (gestures) {
...
+    let press = gestures[0];
+    let wait = gestures[1];
+    let moveTo = gestures[2];
+
+    pressCoordinates = await this.getCoordinates(press);
+    duration = (parseInt(wait.options.ms, 10) / 1000);
+    moveToCoordinates = await this.getCoordinates(moveTo);
+  } else {
+    let longPress = gestures[0];
+    let moveTo = gestures[1];
```
Comment (jlipps) [inline lib/commands/gesture.js]: isn't this a `wait` rather than a `moveTo`, according to the new case above?

**Comment by dkdodson3** [comment]
When will this fix be merged?

**Comment by jlipps** [comment]
just waiting on @imurchie for some answers :-)

**Comment by littor** [comment]
Is this problem fixed already?

**Comment by mykola-mokhnach** [comment]
Closing this PR since the questioned functionality has been already implemented by another one

**Pull request closed without merging**